picard/tagger.py

```diff
@@ -399,6 +408,36 @@ def _open_files_finished(self, target=None, result=None, error=None):
             for file in new_files:
                 file.load(partial(self._file_loaded, target=target))
 
+    def add_directory_list(self, dir_list, parent=None):
+        if parent:
+            parent = os.path.normpath(parent)
+        number_of_dirs = len(dir_list)
+        if number_of_dirs > 1:
+            if parent:
+                self.window.set_statusbar_message(
+                    N_("Adding %(count)d directories from '%(directory)s' ..."),
+                    {'directory': parent, 'count': number_of_dirs}
+                )
+            else:
+                self.window.set_statusbar_message(
+                    N_("Adding %(count)d directories ..."),
+                    {'directory': parent, 'count': number_of_dirs}
+                )
+        else:
+            self.window.set_statusbar_message(
+                N_("Adding directory: '%(directory)s' ..."),
+                {'directory': dir_list[0]}
+            )
+
+        list_of_files = []
+        recursively = config.setting['recursively_add_files']
+        for directory in dir_list:
+            if recursively:
```

Comment:
Yes - use of a method like this is good pythonesque code. However doesn't this code produce a list rather than a list of lists required by _add_files. But I think it only needs a quick tweak to make it work.